src/WebJobs.Script/Host/HostIdValidator.cs

```diff
@@ -0,0 +1,198 @@
+{
+    /// <summary>
+    /// Used to perform Host ID validation checks, ensuring that when hosts are sharing
+    /// a storage account, their computed IDs don't collide.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="ScriptHostIdProvider"/> computes a Host ID and truncates it if needed to
+    /// ensure it's under length limits. For two different Function Apps, this can result in
+    /// both apps resolving to the same Host ID. This can cause problems if those apps share
+    /// a storage account. This class helps detect/prevent such cases.
+    /// </remarks>
+    public class HostIdValidator
+    {
+        public const string BlobPathFormat = "ids/usage/{0}";
+        private const LogLevel DefaultLevel = LogLevel.Warning;
+
+        private readonly IEnvironment _environment;
+        private readonly IAzureStorageProvider _storageProvider;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly IApplicationLifetime _applicationLifetime;
+        private readonly HostNameProvider _hostNameProvider;
+        private readonly ILogger _logger;
+
+        private readonly object _syncLock = new object();
+        private bool _validationScheduled;
+
+        public HostIdValidator(IEnvironment environment, IAzureStorageProvider storageProvider, IApplicationLifetime applicationLifetime,
+            HostNameProvider hostNameProvider, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
+        {
+            _environment = environment;
+            _storageProvider = storageProvider;
+            _serviceProvider = serviceProvider;
+            _applicationLifetime = applicationLifetime;
+            _hostNameProvider = hostNameProvider;
+            _logger = loggerFactory.CreateLogger(LogCategories.HostGeneral);
+        }
+
+        internal bool ValidationScheduled => _validationScheduled;
+
+        public virtual void ScheduleValidation(string hostId, int validationDelaySeconds = 10)
+        {
+            lock (_syncLock)
+            {
+                if (!_validationScheduled)
+                {
+                    // Schedule the validation to run asynchronously after a delay. This delay ensures
+                    // we're not impacting coldstart, and also gives time for the primary host to be
+                    // identified.
+                    Utility.ExecuteAfterDelay(() => Task.Run(() => ValidateHostIdUsageAsync(hostId)), TimeSpan.FromSeconds(validationDelaySeconds));
+                    _validationScheduled = true;
+                }
+            }
+        }
+
+        internal async Task ValidateHostIdUsageAsync(string hostId)
+        {
+            try
+            {
+                if (!_storageProvider.ConnectionExists(ConnectionStringNames.Storage))
+                {
+                    return;
+                }
+
+                var primaryHostStateProvider = _serviceProvider.GetScriptHostServiceOrNull<IPrimaryHostStateProvider>();
```

Comment:
The reason the code is like this is because IPrimaryHostStateProvider is a job host level service, registered [here](https://github.com/Azure/azure-webjobs-sdk/blob/206e194e03d82a101a6b5dffb7b4c3904730801d/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsServiceCollectionExtensions.cs#L90), whereas ScriptHostIdProvider and HostNameProvider are WebHost level services.

Reply:
Removed the IsPrimary check